connectors/spark/src/main/scala/io/unitycatalog/spark/CredPropsUtil.java

```diff
@@ -0,0 +1,262 @@
+    }
+
+    public T tableId(String tableId) {
+      builder.put(UCHadoopConf.UC_TABLE_ID_KEY, tableId);
+      return self();
+    }
+
+    public T tableOperation(TableOperation tableOp) {
+      builder.put(UCHadoopConf.UC_TABLE_OPERATION_KEY, tableOp.getValue());
+      return self();
+    }
+
+    public T path(String path) {
+      builder.put(UCHadoopConf.UC_PATH_KEY, path);
+      return self();
+    }
+
+    public T pathOperation(PathOperation pathOp) {
+      builder.put(UCHadoopConf.UC_PATH_OPERATION_KEY, pathOp.getValue());
+      return self();
+    }
+
+    protected abstract T self();
+
+    public Map<String, String> build() {
+      return builder.build();
+    }
+  }
+
+  public static class S3PropsBuilder extends PropsBuilder<S3PropsBuilder> {
+
+    public S3PropsBuilder() {
+      // Common properties for S3.
+      set("fs.s3a.path.style.access", "true");
+      set("fs.s3.impl.disable.cache", "true");
+      set("fs.s3a.impl.disable.cache", "true");
+    }
+
+    @Override
+    protected S3PropsBuilder self() {
+      return this;
+    }
+  }
+
+  public static class GcsPropsBuilder extends PropsBuilder<GcsPropsBuilder> {
+    @Override
+    protected GcsPropsBuilder self() {
+      return this;
+    }
+  }
+
+  public static class AbfsPropsBuilder extends PropsBuilder<AbfsPropsBuilder> {
+    @Override
+    protected AbfsPropsBuilder self() {
+      return this;
+    }
+  }
+
+  public static Map<String, String> s3FixedCredProps(TemporaryCredentials tempCreds) {
+    AwsCredentials awsCred = tempCreds.getAwsTempCredentials();
+    return new S3PropsBuilder()
+        .set("fs.s3a.access.key", awsCred.getAccessKeyId())
+        .set("fs.s3a.secret.key", awsCred.getSecretAccessKey())
+        .set("fs.s3a.session.token", awsCred.getSessionToken())
+        .build();
+  }
+
+  public static Map<String, String> s3TableTempCredProps(
+      String uri,
+      String token,
+      String tableId,
+      TableOperation tableOp,
+      TemporaryCredentials tempCreds) {
+    AwsCredentials awsCred = tempCreds.getAwsTempCredentials();
+    S3PropsBuilder builder = new S3PropsBuilder()
+        .set(UCHadoopConf.S3A_CREDENTIALS_PROVIDER, AwsVendedTokenProvider.class.getName())
+        .uri(uri)
+        .token(token)
+        .uid(UUID.randomUUID().toString())
+        .credentialType(UCHadoopConf.UC_CREDENTIALS_TYPE_TABLE_VALUE)
+        .tableId(tableId)
+        .tableOperation(tableOp)
+        .set(UCHadoopConf.S3A_INIT_ACCESS_KEY, awsCred.getAccessKeyId())
+        .set(UCHadoopConf.S3A_INIT_SECRET_KEY, awsCred.getSecretAccessKey())
+        .set(UCHadoopConf.S3A_INIT_SESSION_TOKEN, awsCred.getSessionToken());
+
+    // For the static credential case, null expiration time is possible.
+    if (tempCreds.getExpirationTime() != null) {
+      builder.set(UCHadoopConf.S3A_INIT_CRED_EXPIRED_TIME,
+          String.valueOf(tempCreds.getExpirationTime()));
+    }
+
+    builder.set("fs.s3a.access.key", awsCred.getAccessKeyId());
```

Comment:
We don't need to set the three keys anymore, for this provider.  I will remove the 3 lines.